src/core/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,282 @@
+      const focusedIndexInFilteredList = filteredOptions.findIndex(
+        (option) => option === options[focusedIndex],
+      );
+      switch (event.key) {
+        case 'ArrowDown':
+          const nextIndex = Math.min(
+            focusedIndexInFilteredList + 1,
+            filteredOptions.length - 1,
+          );
+          setFocusedIndex(options.indexOf(filteredOptions[nextIndex]));
+          !isOpen && setIsOpen(true); // reopen menu if closed when typing
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'ArrowUp':
+          const previousIndex = Math.max(focusedIndexInFilteredList - 1, 0);
+          setFocusedIndex(options.indexOf(filteredOptions[previousIndex]));
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Enter':
+          setSelectedValue(options[focusedIndex].value);
+          setIsOpen(!isOpen);
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Escape':
+          setIsOpen(false);
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Tab':
+          setIsOpen(false);
+          break;
+        default:
+          !isOpen && setIsOpen(true); // reopen menu if closed when typing
+          break;
+      }
+    },
+    [filteredOptions, focusedIndex, isOpen, options],
+  );
+
+  const menuItems = React.useCallback(
+    (close: () => void) => {
+      return filteredOptions.map((option) => {
+        const isSelected = selectedValue === option.value;
+        const index = options.findIndex(({ value }) => option.value === value);
+        return (
+          <MenuItem
+            id={getOptionId(index)}
+            key={getOptionId(index)}
+            className={cx({ 'iui-focused': focusedIndex === index })}
+            onClick={(value: T) => {
+              setSelectedValue(value);
+              close();
+            }}
+            isSelected={isSelected}
+            ref={(el) => focusedIndex === index && el?.scrollIntoView(false)}
+            role='option'
+            {...option}
+          >
+            {option.label}
+          </MenuItem>
+        );
+      });
+    },
+    [focusedIndex, getOptionId, filteredOptions, selectedValue, options],
+  );
+
+  return (
+    <InputContainer
+      className={className}
+      isIconInline={true}
+      icon={
+        isOpen ? (
+          <SvgCaretUpSmall
+            style={{ cursor: 'pointer' }}
+            onClick={() => setIsOpen(false)}
+          />
+        ) : (
+          <SvgCaretDownSmall
+            style={{ cursor: 'pointer' }}
+            onClick={() => inputRef.current?.focus()}
+          />
+        )
+      }
+      {...rest}
+      id={id}
+    >
+      <Popover
+        content={
+          <Menu
+            id={`${id}-list`}
+            className='iui-scroll'
+            style={{ minWidth, maxHeight: '300px' }}
```

Comment:
You should do the same as in #330 :
![image](https://user-images.githubusercontent.com/36186912/134938725-fffb53ab-1744-4f42-b978-aa5db675a96a.png)

Reply:
I am not sure about this. It's using `document` instead of `ownerDocument`. I would have just used the CSS `min` function with `100vw`.